qupath-core/src/main/java/qupath/lib/objects/utils/ObjectMerger.java

```diff
@@ -189,15 +190,15 @@ private List<List<PathObject>> computeClustersIterative(Collection<? extends Pat
                     else
                         allPotentialNeighbors = allObjects;
                     var neighbors = filterCompatibleNeighbors(current, allPotentialNeighbors);
-                    for (var neighbor : neighbors) {
+                    neighbors.parallelStream().forEach(neighbor -> {
                         if (!alreadyVisited.contains(neighbor)) {
```

Comment:
`alreadyVisited` is not thread-safe, so I would move this line outside of the concurrent block (which shouldn't have any real performance issue?)